README.md

```diff
@@ -147,6 +147,57 @@ resources:
         newKey: user_id
 ```
 
+### Batch Resources with `properties` paramerters
+
+Instead of accepting just a list of users (`user_ids`), a batch resource could accept both a list of users (`user_ids`) and a list of properties (`properties`) to fetch about that user:
+
+```js
+const getUserInfo = (args: { user_ids: Array<number>, properties: Array<string> }): Promise<Array<UserInfo>> =>
+    fetch('/userInfo', args);
+
+const users = getUserInfo({
+  user_ids: [1, 2, 3],
+  properties: ['firstName', 'age']
+});
+
+e.g. users =>
+[
+  { id: 1, firstName: 'Alice', age: 42 },
+  { id: 2, firstName: 'Bob', age: 70 },
+  { id: 3, firstName: 'Carol', age: 50 },
+]
+```
+
+To batch up calls to this resource with different `properties` for different `user_ids`, we specify `propertyBatchKey` in the config to describe the "properties" argument.
+We specify `responseKey` in the config as the key in the response objects corresponds to `batchKey`.
+
+The config for our `getUserInfoV2` would look like this:
+
+```yaml
+resources:
+    getUserInfo:
+        isBatchResource: true
+        batchKey: user_ids
+        newKey: user_id
+        propertyBatchKey: properties
+        responseKey: id
+```
+
+**IMPORTANT NOTE**
+To use this feature, there are several restrictions. (Please open an issue if you're interested in helping us support other use cases):
+
+1. The resource accept two non-optional `list` type parameters, like this:
+
+```
```

Comment:
You can preview what this looks like here: https://github.com/Yelp/dataloader-codegen/blob/344462dc8d534f5f3510b6cff4e46bf87bd8784f/README.md#batch-resources-with-properties-paramerters (or there's vscode plugins to preview the rendered markdown live)

this probably should be indented inline with the above bullet